keystone_json_assignment/json.py

```diff
@@ -185,6 +185,22 @@ def list_role_assignments(self, role_id=None,
             # so just return the SQL grants
             return role_assignments
         for user, projects in self.userprojectmap.items():
+            expected_role_assignments = []
+            for project in projects:
+                if project_ids and project not in project_ids:
+                    # We're filtering on projects and this one isn't one of
+                    # them, move on
+                    continue
+                expected_role_assignments.append({
+                    'role_id': self.role_id,
+                    'user_id': user_id,
```

Comment:
This user id is converted on line 212, creating the role assignments prematurely will cause the wrong user ids to get into the list.

Reply:
thank you, fixed